### PR TITLE
Support for official mysql 8 packages

### PR DESCRIPTION
**Pull request by waynegemmell**
I've added support for the official mysql 8.0 packages and the relevant repositories.
Package names are mysql-community-server and mysql-community-client.
I know the repo setting breaks convention a bit but it is compulsory so I'm not sure how else to properly include it.

This change requires the salt pull request I created https://github.com/saltstack/salt/pull/51240 to be able to grant users.

### CODE REVIEW

File: mysql/server.sls [hunk 1]
```diff
@@ -31,18 +44,32 @@ mysql_debconf:
...
+        'mysql-server/default-auth-override': {'type': 'string', 'value':'Use Legacy Authentication Method (Retain MySQL 5.x Compatibility)'}
+    - require_in:
+      - pkg: {{ mysql.serverpkg }}
+    - require:
+      - pkg: mysql_debconf_utils
+{% else %}
 mysql_password_debconf:
   debconf.set:
-    - name: mysql-server
+    - name: {{ mysql.serverpkg }}
```
Comment (aboe76) [inline mysql/server.sls]: wasn't there an issue with mariadb-server which has the same debconf settings as mysql-server?

**Comment by noelmcloughlin** [comment]
In https://github.com/saltstack-formulas/mysql-formula/pull/186 I added standardized mapping files but unfortunately skipped the `codenamemap.yaml`.  This PR suggests its needed.

Hi @waynegemmell any chance you could introduce a *codenamemap.yaml* for this PR?

**Comment by waynegemmell** [comment]
I'll have a look at the codenamemap.yaml. Not sure what that entails. I'll let you know if I can't come right.

File: mysql/server.sls (hunk 1 above)
Comment (waynegemmell) [reply mysql/server.sls]: I'll revert that change, it wasn't really necessary after I realised that mysql-community-server has it's own set of debconf settings.

**Comment by waynegemmell** [comment]
Sorry, can you please explain the purpose of the  codenamemap.yaml file in this context?

**Comment by noelmcloughlin** [comment]
I understand the purpose of the codenamemap is to handle OS release specific differences. So if you need to handle an official repo from some distro this method scales best.
```
                merge=salt['grains.filter_by'](
                    osfamilymap,
                    grain='os_family',
                    merge=salt['grains.filter_by'](
                        oscodenames,
                        grain='oscodename',
                        merge=salt['pillar.get']
```
Here are three examples of implementations people done.
```
$ for f in $( find . | grep codenamemap ); do echo $f;cat $f; done


./docker-formula/docker/codenamemap.yaml
# vim: sts=2 ts=2 sw=2 et ai

wheezy:
  kernel:
    pkg:
      name: linux-image-amd64
      fromrepo: wheezy-backports
    pkgrepo:
      name: deb http://http.debian.net/debian wheezy-backports main
      humanname: Wheezy Backports
      dist: wheezy-backports
    pkgs: []

jessie:
  kernel:
    pkg:
      name: linux-image-amd64
      fromrepo: jessie-backports
    pkgrepo:
      name: deb http://http.debian.net/debian jessie-backports main
      humanname: Jessie Backports
      dist: jessie-backports
    pkgs: []

trusty:
  kernel:
    pkgs:
      - linux-image-extra-virtual
      - linux-image-extra-{{ grains.kernelrelease }}

precise:
  kernel:
    pkgs:
      - linux-image-generic-lts-raring
      - linux-headers-generic-lts-raring

./postgres-formula/postgres/codenamemap.yaml
### Set parameters based on PostgreSQL version supplied with particular distro

{% import_yaml "postgres/repo.yaml" as repo %}
  {# Generate lookup dictionary map for OS and derivative distributions
    name: distro codename
    version: PostgreSQL release version
    codename: optional grain value if `name` does not match the one returned by `oscodename` grain
  #}

{% macro debian_codename(name, version, codename=none) %}

  {% if repo.use_upstream_repo == true %}
    {% set version = repo.version %}
    {% set fromrepo = repo.fromrepo|default(name ~ '-pgdg', true) %}
  {% else %}
    {% set fromrepo = name %}
  {% endif %}
  {% set cluster_name = repo.cluster_name %}
  {% set conf_dir = '/etc/postgresql/{0}/{1}'.format(version, cluster_name) %}
  {% set data_dir = '/var/lib/postgresql/{0}/{1}'.format(version, cluster_name) %}

{{ codename|default(name, true) }}:
  # PostgreSQL packages are mostly downloaded from `main` repo component
  conf_dir: {{ conf_dir }}
  data_dir: {{ data_dir }}
  fromrepo: {{ fromrepo }}
  pkg_repo:
    name: 'deb http://apt.postgresql.org/pub/repos/apt {{ name }}-pgdg main'
  pkg: postgresql-{{ version }}
  pkg_client: postgresql-client-{{ version }}
  prepare_cluster:
    pgcommand: pg_createcluster {{ version }} {{ cluster_name }} -d
    user: root

{% endmacro %}


{% macro fedora_codename(name, version, codename=none) %}

  {# use upstream version if configured #}
  {% if repo.use_upstream_repo == true %}
    {% set version = repo.version %}
  {% endif %}

{{ codename|default(name, true) }}:
  # PostgreSQL packages are mostly downloaded from `main` repo component
  pkg_repo:
    baseurl: 'https://download.postgresql.org/pub/repos/yum/{{ version }}/fedora/fedora-$releasever-$basearch'

{% endmacro %}

## Debian GNU/Linux
{{ debian_codename('wheezy', '9.1') }}
{{ debian_codename('jessie', '9.4') }}
{{ debian_codename('stretch', '9.6') }}

# `oscodename` grain has long distro name
# if `lsb-release` package not installed
{{ debian_codename('wheezy', '9.1', 'Debian GNU/Linux 7 (wheezy)') }}
{{ debian_codename('jessie', '9.4', 'Debian GNU/Linux 8 (jessie)') }}
{{ debian_codename('stretch', '9.6', 'Debian GNU/Linux 9 (stretch)') }}

## Ubuntu
{{ debian_codename('trusty', '9.3') }}
{{ debian_codename('xenial', '9.5') }}
{{ debian_codename('artful', '9.6') }}
{{ debian_codename('bionic', '10') }}
{{ debian_codename('cosmic', '10') }}

## Fedora
# `oscodename` grain has long distro name
{{ fedora_codename('Fedora-27', '9.6', 'Fedora 27 (Twenty Seven)') }}
{{ fedora_codename('Fedora-26', '9.6', 'Fedora 26 (Twenty Six)') }}

# vim: ft=sls

./tomcat-formula/tomcat/codenamemap.yaml
trusty:
  ver: 7
  pkg: tomcat7
  manager_pkg: tomcat7-admin
  conf_dir: /etc/tomcat7
  main_config: /etc/default/tomcat7
  main_config_template: salt://tomcat/files/tomcat-default-Debian.template
  service: tomcat7
  user: tomcat7
  group: tomcat7
wheezy:
  pkg: tomcat7
  manager_pkg: tomcat7-admin
  conf_dir: /etc/tomcat7
  main_config: /etc/default/tomcat7
  main_config_template: salt://tomcat/files/tomcat-default-Debian.template
  service: tomcat7
  user: tomcat7
  group: tomcat7

```

**Comment by noelmcloughlin** [comment]
I fixed the conflicts and merging. This PR is causing a backlog.

**Pull request merged**